docs/install/ubuntu_setup.md

```diff
@@ -1,67 +1,103 @@
+<hr>
 
-MXNet currently supports Python, R, Julia, Scala, and Perl. For users of R on Ubuntu operating systems, MXNet provides a set of Git Bash scripts that installs all of the required MXNet dependencies and the MXNet library.
+## Contents
 
-The simple installation scripts set up MXNet for Python and R on computers running Ubuntu 12 or later. The scripts install MXNet in your home folder ```~/mxnet```.
+* [CUDA Dependencies](#cuda-dependencies)
+* [Quick Installation](#quick-installation)
+    * [Python](#install-mxnet-for-python)
+    * [pip Packages](#pip-package-availability)
+* [Standard Installation](#standard-installation)
+* [Installing Language Packages](#installing-language-packages-for-mxnet)
+    * [R](#install-the-mxnet-package-for-r)
+    * [Julia](#install-the-mxnet-package-for-julia)
+    * [Scala](#install-the-mxnet-package-for-scala)
+    * [Perl](#install-the-mxnet-package-for-perl)
+  * [Contributions](#contributions)
+  * [Next Steps](#next-steps)
 
-## Prepare environment for GPU Installation
+<hr>
 
-If you plan to build with GPU, you need to set up the environment for CUDA and CUDNN.
+## CUDA Dependencies
 
-First, download and install [CUDA 8 toolkit](https://developer.nvidia.com/cuda-toolkit).
+If you plan to build with GPU, you need to set up the environment for CUDA and cuDNN.
 
-Then download [cudnn 6](https://developer.nvidia.com/cudnn).
+First, download and install [CUDA toolkit](https://developer.nvidia.com/cuda-toolkit). CUDA 9.2 is recommended.
 
-Unzip the file and change to the cudnn root directory. Move the header and libraries to your local CUDA Toolkit folder:
+Then download [cuDNN 7.1.4](https://developer.nvidia.com/cudnn).
+
+Unzip the file and change to the cuDNN root directory. Move the header and libraries to your local CUDA Toolkit folder:
 
 ```bash
-    tar xvzf cudnn-8.0-linux-x64-v6.0.tgz
+    tar xvzf cudnn-9.2-linux-x64-v7.1
     sudo cp -P cuda/include/cudnn.h /usr/local/cuda/include
     sudo cp -P cuda/lib64/libcudnn* /usr/local/cuda/lib64
     sudo chmod a+r /usr/local/cuda/include/cudnn.h /usr/local/cuda/lib64/libcudnn*
     sudo ldconfig
 ```
 
-Finally, add configurations to config.mk file:
+<hr>
 
-```bash
-    cp make/config.mk .
-```
 
 ## Quick Installation
 
-### Install MXNet for R
+### Install MXNet for Python
 
-MXNet requires R-version to be 3.2.0 and above. If you are running an earlier version of R, run below commands to update your R version, before running the installation script.
+#### Dependencies
 
-```bash
-    sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E084DAB9
-    sudo add-apt-repository ppa:marutter/rdev
+The following scripts will install Ubuntu 16.04 dependencies for MXNet Python development.
 
-    sudo apt-get update
-    sudo apt-get upgrade
-    sudo apt-get install r-base r-base-dev
+```bash
+wget https://raw.githubusercontent.com/apache/incubator-mxnet/master/ci/docker/install/ubuntu_core.sh
```

Comment:
Puh, I'd prefer if we keep these scripts internal and not user-facing. We install a bunch of crap in there because it's required for some rare use-case. Also, they might be changed at any time. It would be better if we keep these two concerns separate and let the installation guide test ensure that everything works properly.

Reply:
I was thinking about maintainability. We essentially test these constantly. But, I can agree with the bloat issue. What if we get a stripped down version just for installation instructions, and add that to CI?
Then I can work towards removing the dependency on the install page being parsed for a CI test - something that can end up being brittle and doesn't really thoroughly test as it's pretty much just doing pip installs.

Reply:
Sure, feel free to create a new nightly test. You could then reference the resources of that test in the installation instructions.